src/lib/ledger_catchup/normal_catchup.ml

```diff
@@ -600,20 +601,14 @@ let verify_transitions_and_build_breadcrumbs ~logger
 
 let garbage_collect_subtrees ~logger ~subtrees =
   List.iter subtrees ~f:(fun subtree ->
-      ignore
-        ( Rose_tree.map subtree ~f:Cached.invalidate_with_failure
-          : 'a Rose_tree.t ) ) ;
+      Rose_tree.iter subtree ~f:(fun (node, _vc) ->
+          (* TODO reject callback? *)
```

Comment:
Depends on the reason. We call this function in 2 different contexts. In the context where we are shutting down this process, we just want to fill the validation callbacks with `Ignore`. For the case of handling an error in verifying and building breadcrumbs, we need to handle the errors accordingly.